config/values.yml

```diff
@@ -1,8 +1,21 @@
-#@data/values
+#@data/values-schema
 ---
+#@schema/desc "The namespace in which to deploy secretgen-controller"
 namespace: secretgen-controller
+#@schema/desc "Whether to create namespace specified for secretgen-controller"
 create_namespace: true
-
-push_images: false
-image_repo: docker.io/k14s/secretgen-controller-test
 secretgen_controller_version: "develop"
```

Comment:
We maybe? don't want this in our public api for this values file.

This values file will be part of the bundle, but the problem lies https://github.com/vmware-tanzu/carvel-secretgen-controller/pull/65/files#diff-1f0ed1aa0e177024e3bfc3a58979f9d63c23921a83c8e9aa6a6b3b269f728c06L9

We need the version to be a value for our config, or else we would need to hard code it in each time / have automation that edits `config/deployment.yml` each time. This is a stick value, maybe it's fine to expose? But we don't want it to be configurable, or else the user could end up with a difference between the annotation in the deployment, and the actual version of secretgen-controller.

Reply:
it looks like it gets compiled into the binary in the dockerfile:
`28:RUN CGO_ENABLED=0 GOOS=linux go build -mod=vendor -ldflags="-X 'main.Version=$SGCTRL_VER' -buildid=" -trimpath -o controller ./cmd/controller/...`

and also set as an annotation in the deployment:
```
config/deployment.yml
8:  annotations:
9:    secretgen-controller.carvel.dev/version: #@ data.values.secretgen_controller_version
```


I think we were talking to some other folks about packaging this morning who were saying they basically do two templating passes:  a first one to lock things down before the package is made,  and then a second one that's part of the packaging process.     But i think that amounts to what you describe above as "have automation that edits config/deployment.yml"